nemo/collections/nlp/data/language_modeling/megatron/retro_dataset.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RETRO Style dataset."""
+
+import torch
+
+from nemo.collections.nlp.data.language_modeling.megatron.base_dataset_utils import get_train_valid_test_split_
+from nemo.collections.nlp.data.language_modeling.megatron.megatron_dataset import MegatronDataset
+from nemo.utils import logging
+
+# from nemo.collections.nlp.data.language_modeling.megatron.dataset_utils import (
+#     create_masked_lm_predictions,
+#     create_tokens_and_tokentypes,
+#     get_a_and_b_segments,
+#     get_samples_mapping,
+#     truncate_segments,
+# )
+# from nemo.collections.nlp.data.language_modeling.megatron.indexed_dataset import MMapIndexedDataset
+try:
+    from apex.transformer import parallel_state
+
+    HAVE_APEX = True
+except (ImportError, ModuleNotFoundError):
+    HAVE_APEX = False
+
+
+class MockRETRODataset(MegatronDataset):
+    def __init__(self, cfg, trainer, tokenizer, name, size):
+        super().__init__(cfg, trainer=trainer)
+        self.name = name
+        self.tokenizer = tokenizer
+        self._cfg = cfg
+        self.size = size
+        seed_val = parallel_state.get_data_parallel_rank() * 131 + 97
+        torch.manual_seed(seed_val)
```

Comment:
This could be set in other places as well?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/0fe2ae74e919c8ef220a029cab2203572ca536a5/pytorch_lightning/utilities/seed.py#L75

Reply:
Won't this call overwrite the seed?

Reply:
Also, gets called here: https://github.com/NVIDIA/NeMo/blob/ddd87197e94ca23ae54e641dc7784e64c00a43d6/nemo/collections/nlp/modules/common/megatron/megatron_init.py#L120

Reply:
yeah. It is only useful for testing. I will replace it with real RETRO dataset later so seed number has no effects